File: Server/Entities/Events.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace Server.Entities
{
    public sealed class Events
    {
        public int Id { get; set; }
        public required string Title { get; set; }
        [Column("Date_Of_Event")]
        public DateOnly DateOfEvent { get; set; }
        [Column("Time_Of_Event")]
        public TimeOnly TimeOfEvent { get; set; }
        public required string Category { get; set; }    
        public required string Location { get; set; }
        public string? Description { get; set; }

        [Column("Created_By")]
        public required int CreatedBy { get; set; }
    }
}

File: Client/src/Types/Events.ts
// eslint-disable-next-line @typescript-eslint/no-namespace
export namespace Events {
  export interface EventRequest {
    Title: string;
    DateOfEvent: string;
    TimeOfEvent: string;
    Category: string;
    Location: string;
    Description?: string;
  }

  export interface EventResponse {
    id: number;
     title: string;
    dateOfEvent: Date;
    timeOfEvent: string;
    category: string;
    location: string;
    description?: string;
    count: number;
  }
}
